### ISSUE TITLE
[Voice Access] TextBox & NumberBox (Inline mode) x button cannot be targetted

### ISSUE THREAD

**Issue by carlos-zamora** [microsoft/microsoft-ui-xaml#7996]
### Describe the bug

A `NumberBox` with `SpinButtonPlacementMode: inline` provides an 'x' button that clears the text when it is being interacted with. However, when using voice access, the 'x' button cannot be targeted.

### Steps to reproduce the bug

1. Open WinUI 2 Gallery > NumberBox
2. Change SpinButton placement to Inline
3. Click on the text area in the "Enter an integer" number box
4. Open Voice Access
5. Say "Show numbers"
6. Observe that 25, 26, and 27 appear on the number box (respective to input area, up arrow, and down arrow)

### Expected behavior

The x button should have a number tag on it

### NuGet package version

None

### Windows version

Windows 11 (22H2): Build 22621

**Comment by HariniMalothu17** [comment]
This is a bydesign scenario as the user can clear the text another way through Voice Access, using commands like select all and clear, which has the same functionality as the close button.Hence closing the issue
